### PR TITLE
feat: stm32mp kirkstone layer

### PR DESCRIPTION
**Pull request by SergeyFabric**
this layer modified from dunfel.
there still work to do because the bootfs.image_gptimg.env file missing for building the final image. may be there another problems as well.

Changelog: Commit
Ticket: 5331

### CODE REVIEW

File: meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    cp ${DEPLOY_DIR_IMAGE}/kernel/${KERNEL_DEVICETREE} ${WORKDIR}/bootfs/
+
+    # Remove file generated by mender-part-images.bbclass
+    rm "$wks"
+
+    cat >> "$wks" <<EOF
+
+part metadata1 --source rawcopy --fsoptions "noauto" --part-name=metadata1 --sourceparams="file=${DEPLOY_DIR_IMAGE}/arm-trusted-firmware/metadata.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 512K --align 420
+part metadata2 --source rawcopy --fsoptions "noauto" --part-name=metadata2 --sourceparams="file=${DEPLOY_DIR_IMAGE}/arm-trusted-firmware/metadata.bin" --ondisk mmcblk --part-type 0x8301 --fixed-size 512K 
+part fip-a  --source rawcopy --fstype=ext4 --fsoptions "noauto" --part-name=fip-a --sourceparams="file=${DEPLOY_DIR_IMAGE}/fip/fip-stm32mp153a-ground4-mx-trusted.bin" --ondisk mmcblk --part-type 19d5df83-11b0-457b-be2c-7559c13142a5 --uuid="4FD84C93-54EF-463F-A7EF-AE25FF887087" --fixed-size 4096K
```
Comment (TheYoctoJester) [inline meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass:21]: these two lines are probably leftovers from a custom board, and fail the standard build

File: meta-mender-st-stm32mp/templates/local.conf.append [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+# # Upstream PR: https://github.com/STMicroelectronics/meta-st-stm32mp/pull/11
+PREFERRED_PROVIDER_u-boot="u-boot-stm32mp"
+PREFERRED_PROVIDER_virtual/bootloader="u-boot-stm32mp"
+
+# # Add custom image class that appends STM32MP bootloader to the gptimg
+IMAGE_CLASSES += "stm32mp-gptimg"
+
+
+MENDER_BOOT_PART = "${MENDER_STORAGE_DEVICE_BASE}6"
```
Comment (TheYoctoJester) [inline meta-mender-st-stm32mp/templates/local.conf.append:53]: these numbers need to be checked once the custom board leftovers are removed

File: meta-mender-st-stm32mp/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+The supported and tested boards are:
+
+- [STM32MP157C Discovery Kit](https://hub.mender.io/t/stm32mp157c-discovery-kit/1676)
+
+Visit the individual board links above for more information on status of the
+integration and more detailed instructions on how to build and use images
+together with Mender for the mentioned boards.
+
+## Dependencies
```
Comment (TheYoctoJester) [inline meta-mender-st-stm32mp/README.md:13]: all dependencies are on dunfell, and as we discussed at https://hub.mender.io/t/stm32mp1-yocto-kirkstone-issues/5331 the setup instructions should actually refer to the ST wiki. I can also create a new Mender Hub article and retire the old one, free to suggestions here and happy to help

**Review by TheYoctoJester — CHANGES_REQUESTED** [review]
Great work! Needs just some minor fixes:
- remove leftover custom board firmware
- add SoB

For the documentation/readme, this I can help with if desired.

File: meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass (hunk 1 above)
Comment (SergeyFabric) [reply meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass]: I changed it to the right board. it should be in the gpt image.

File: meta-mender-st-stm32mp/templates/local.conf.append (hunk 2 above)
Comment (SergeyFabric) [reply meta-mender-st-stm32mp/templates/local.conf.append:53]: this one ok. according gptimg class

File: meta-mender-st-stm32mp/README.md (hunk 3 above)
Comment (SergeyFabric) [reply meta-mender-st-stm32mp/README.md:22]: I added some changes to the readme. please check it.

**Comment by TheYoctoJester** [comment]
@SergeyFabric can you add your signed off by do I can merge the state? I keep on running into rough edges here and there but I blame that on my lack of experience on stm32 and so we can move forward.

**Comment by SergeyFabric** [comment]
I never used this feature and do not know how to sign off all the commits.

**Review by TheYoctoJester — APPROVED** [review]

**Comment by TheYoctoJester** [comment]
All look good except a7a1bd006aaa6f1206516dc16d781526aff4d132! :-) Can you try -s on that one too? If it fails for some reason let me know, then I'll see if I can skip the check.

**Comment by SergeyFabric** [comment]
I don't know how 😒

**Comment by TheYoctoJester** [comment]
Ah! All good. It should work like this.
1) checkout the commit to a new branch, like `git checkout -b signoff a7a1bd0`
2) then do a amend commit to sign off: `git commit -s --amend` - don't need to change the message otherwise.
3) change to your original branch: `git checkout kirkstone-next-stm32mp`
4) rebase that onto the signed one: `git rebase signoff`
5) push `git push -f`

That should effectively sign the one commit and replace the branch that I can merge with the new one. the `signoff` branch can be deleted after the rebase.

If this doesn't work, then I'll find a way. Thanks!

**Comment by SergeyFabric** [comment]
Done tnx for the help.

**Comment by SergeyFabric** [comment]
Tnx , Done

**Pull request merged**